src/protocol.h

```diff
@@ -145,8 +145,32 @@ class CInv
     {
         int nVersion = s.GetVersion();
         READWRITE(type);
+
+        // The implicit P2P network protocol inherited from Bitcoin Core has
+        // zcashd nodes sort-of ignoring unknown CInv message types in inv
+        // messages: they are added to the known transaction inventory, but
+        // AlreadyHave returns true, so we do nothing with them. Meanwhile for
+        // getdata messages, ProcessGetData ignores unknown message types
+        // entirely.
+        switch (type) {
+        case MSG_TX:
+        case MSG_BLOCK:
+        case MSG_FILTERED_BLOCK:
+            break;
+        case MSG_WTX:
+            if (nVersion < CINV_WTX_VERSION) {
+                throw std::ios_base::failure(
+                    "Negotiated protocol version does not support CInv message type MSG_WTX");
+            }
+            break;
+        default:
+            // This includes UNDEFINED, which should never be serialized.
+            throw std::ios_base::failure("Unknown CInv message type");
+        }
```

Comment:
If we switched to the weaker version (only reject unknown CInv message types if 170014 is negotiated), we'd make this change:
```suggestion
        switch (type) {
        case MSG_TX:
        case MSG_BLOCK:
        case MSG_FILTERED_BLOCK:
        case MSG_WTX:
            break;
        default:
            // This includes UNDEFINED, which should never be serialized.
            if (nVersion >= CINV_WTX_VERSION) {
                throw std::ios_base::failure("Unknown CInv message type");
            }
        }
```

Reply:
We decided to stick with the stronger change.